packages/ai/core/generate-text/generate-text.test.ts

```diff
@@ -783,3 +783,109 @@ describe('tools with custom schema', () => {
     ]);
   });
 });
+
+
+
+describe('result.text.timeout', () => {
```

Comment:
can the abort signal function be injected / stubbed to make the tests simpler and deterministic? there should not be any date now or delay ideally. eg generate text already has similar test functions like date now that are passed in